openhsi/cameras.py

```diff
@@ -331,4 +331,56 @@ def get_temp(self) -> float:
     def get_mac(self)-> str:
         return ':'.join(['{}{}'.format(a, b)
                          for a, b
-                         in zip(*[iter('{:012x}'.format(cam.deviceSettings['GevMACAddress'].value))]*2)])
\ No newline at end of file
+                         in zip(*[iter('{:012x}'.format(cam.deviceSettings['GevMACAddress'].value))]*2)])
+
+# Cell
+from typing import Tuple
+
+@delegates()
+class FlirCamera(OpenHSI):
+    """Interface for FLIR camera"""
+
+    def __init__(self, **kwargs):
+        """Initialise FLIR camera"""
+        super().__init__(**kwargs)
+
+        try:
+            from simple_pyspin import Camera
+        except ModuleNotFoundError:
+            warnings.warn("ModuleNotFoundError: No module named 'PySpin'.",stacklevel=2)
+
+        self.flircam = Camera()
+        self.flircam.GainAuto = 'Off'
+        self.flircam.Gain = 0
+        self.flircam.AcquisitionFrameRateAuto = 'Off'
+        self.flircam.AcquisitionFrameRateEnabled = True
+        self.flircam.AcquisitionFrameRate = int( 1_000/self.settings["exposure_ms"] )
+
+        self.flircam.ExposureAuto = 'Off'
+        self.flircam.ExposureTime = self.settings["exposure_ms"]*1e3 # convert to us
```

Comment:
update_intsphere_cube need to change and query the exposure (and thus frame rate)
- will eventually need set_exposure method (which should then update settings["exposure_ms"] with value from camera).  Often if ask for 10ms, you actually get something like 9.9ms (depends on camera etc).

Reply:
Will be in my next commit